hazelcast/src/test/java/com/hazelcast/map/impl/eviction/ExpirationManagerTimeoutTest.java

```diff
@@ -36,21 +36,21 @@
 public class ExpirationManagerTimeoutTest extends HazelcastTestSupport {
 
     @Test
-    public void afterShortExpirationEntryShouldBeAway() throws InterruptedException {
+    public void locking_does_not_cause_expired_keys_live_forever() {
         final String KEY = "key";
 
         final HazelcastInstance node = createHazelcastInstance();
         try {
             IMap<String, String> map = node.getMap("test");
             // after 1 second entry should be evicted
-            map.set(KEY, "value", 1500, TimeUnit.MILLISECONDS);
+            map.set(KEY, "value", 4, TimeUnit.SECONDS);
             // short time after adding it to the map, all ok
             map.lock(KEY);
             Object object = map.get(KEY);
             map.unlock(KEY);
             assertNotNull(object);
```

Comment:
One possible improvement is that we perform this assertion only if the time elapsed since `map.set` is <4s. That makes the assertion ignorant to bigger hiccups (with a high chance at least), while we still do it under normal conditions.